frontend/packages/dev-console/src/components/formik-fields/InputSearchField.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import cx from 'classnames';
+import { useField } from 'formik';
+import { ControlLabel, FormControl, FormGroup, HelpBlock, InputGroup } from 'patternfly-react';
+import { SearchInputFieldProps } from './field-types';
+import { getValidationState } from './field-utils';
+
+const InputSearchField: React.FC<SearchInputFieldProps> = ({
+  label,
+  helpText,
+  onSearch,
+  ...props
+}) => {
+  const [field, { touched, error }] = useField(props.name);
+  return (
+    <FormGroup
+      controlId={`${props.name}-field`}
+      validationState={getValidationState(error, touched)}
+    >
+      <ControlLabel className={cx({ 'co-required': props.required })}>{label}</ControlLabel>
+      <InputGroup>
+        <FormControl
+          {...field}
+          {...props}
+          aria-describedby={helpText && `${props.name}-help`}
+          onKeyDown={(e: KeyboardEvent) => e.keyCode === 13 && onSearch(field.value)}
```

Comment:
If you add this here, you can remove the `onKeyDown` event handler from `DeployImageForm` and it will be more consistent with other forms.
```suggestion
         onKeyDown={(e: KeyboardEvent) => {
            if (e.keyCode === 13) {
              e.preventDefault();
              e.stopPropagation();
              onSearch(field.value);
            }
          }}   
```

Reply:
use `e.key` instead of `e.keyCode`?

Reply:
good idea

Reply:
After trying this change, it did not work. @rohitkrai03 maybe you can play with it after it is merged.

Reply:
I did try it while reviewing yesterday and suggested this change because it worked for me.

Reply:
You can send me a PR to the branch and I'll include it if it works. :)

Reply:
Let's refactor it in a separate PR with a bunch of other things that we talked about in this PR.

Reply:
sounds good to me